lib/matrix.gi

```diff
@@ -2348,9 +2364,9 @@ InstallOtherMethod( RankMat,
 ##
 #M  SemiEchelonMat( <mat> )
 ##
-InstallMethod( SemiEchelonMatDestructive,
+InstallOtherMethod( SemiEchelonMatDestructive,
     "generic method for matrices",
-    [ IsMatrix and IsMutable ],
+    [ IsMatrixOrMatrixObj and IsMutable ],
```

Comment:
I think rather the declaration of `SemiEchelonMatDestructive` should be changed to expect `IsMatrixOrMatrixObj and IsMutable`.

Reply:
I would be happy with that.